.github/workflows/functional-test.yaml

```diff
@@ -559,6 +560,30 @@ jobs:
             --subscription ${{ secrets.INTEGRATION_TEST_SUBSCRIPTION_ID }} \
             --name ${{ env.AZURE_TEST_RESOURCE_GROUP }} \
             --yes --verbose
+      # TODO add once tested: if: failure() && steps.publish-tf-recipes.outcome == 'failure' 
+      - name: Get pod logs for recipe publishing failure
+        id: get-pod-logs
+        run: |
+          # Create pod-logs directory
+          mkdir -p recipes/pod-logs
+          # Get pod logs and save to file
+          namespace="radius-test-tf-module-server"
+          label="app.kubernetes.io/name=tf-module-server"
+          pod_names=($(kubectl get pods -l $label -n $namespace -o jsonpath='{.items[*].metadata.name}'))
+          for pod_name in "${pod_names[@]}"; do
+            kubectl logs $pod_name -n $namespace > recipes/pod-logs/${pod_name}.txt
```

Comment:
We can add a log here like `echo "Pod ${pod_name} logs saved to recipes/pod-logs/"`

Reply:
There's a line below that logs when all of them are uploaded but will update to this